common/init.sls

```diff
@@ -1,11 +1,38 @@
 {% from tpldir ~ '/map.jinja' import config with context %}
 
 {% if grains['kernel'] != 'Darwin' %}
-python-pkgs:
+# Ubuntu has python2 as default python
+python2:
   pkg.installed:
     - pkgs:
-      - python-pip
+      - python
       - python-dev
+
+python3:
+  pkg.installed:
+    - pkgs:
+      - python3
+
+# Ensure pip is default by purging pip3
+pip:
+  pkg.installed:
+    - pkgs:
+      - python-pip
+    - reload_modules: True
+
+pip3:
+  pkg.purged:
+    - pkgs:
+      - python3-pip
+
+# Virtualenv package creates virtualenv and virtualenv-3.4 executables
```

Comment:
Are the default python/pip/virtualenv the 2.7 ones?

`bin_env` is for both selecting a virtualenv or selecting a custom binary to run. It lets you say "Install this pip package, but use the `pip3.4` binary". https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.pip.html